src/lib/configs/project/dockerCompose.ts

```diff
@@ -0,0 +1,315 @@
+        FLUENCE_SYSTEM_SERVICES__ENABLE: "aqua-ipfs,decider",
+        FLUENCE_ENV_CONNECTOR_WALLET_KEY:
+          "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f",
+        FLUENCE_ENV_CONNECTOR_CONTRACT_ADDRESS:
+          "0x0f68c702dC151D07038fA40ab3Ed1f9b8BAC2981",
+        FLUENCE_SYSTEM_SERVICES__DECIDER__DECIDER_PERIOD_SEC: 10,
+        FLUENCE_MAX_SPELL_PARTICLE_TTL: "9s",
+        FLUENCE_SYSTEM_SERVICES__DECIDER__NETWORK_ID: 31337,
+        FLUENCE_ROOT_KEY_PAIR__PATH: `/run/secrets/${name}`,
+      },
+      command: [
+        "--aqua-pool-size=5",
+        `-t=${tcpPort}`,
+        `-w=${webSocketPort}`,
+        "--external-maddrs",
+        `/dns4/${name}/tcp/${tcpPort}`,
+        `/dns4/${name}/tcp/${webSocketPort}/ws`,
+        "--allow-private-ips",
+        tcpPort === TCP_PORT_START
+          ? "--local"
+          : `--bootstraps=/dns/${bootStrapNoxName}/tcp/${TCP_PORT_START}`,
+      ],
+      depends_on: [IPFS_CONTAINER_NAME],
+      secrets: [name],
+    },
+  ];
+}
+
+async function genDockerCompose(
+  providerConfig: ProviderConfigReadonly,
+): Promise<LatestConfig> {
+  const secretKeys = await genSecretKeysOrReturnExisting(
+    Object.keys(providerConfig.computePeers),
+  );
+
+  const [bootStrapNoxNameAndKey, ...restSecretKeys] = secretKeys;
+
+  assert(
+    bootStrapNoxNameAndKey !== undefined,
+    "At least one nox is required to generate a docker-compose.yml",
+  );
+
+  const { name: bootStrapNoxName } = bootStrapNoxNameAndKey;
+
+  return {
+    version: "3",
+    services: {
+      [CHAIN_CONTAINER_NAME]: {
+        image: versions.chain,
+        pull_policy: "always",
+        ports: [`${CHAIN_PORT}:${CHAIN_PORT}`],
+      },
+      [IPFS_CONTAINER_NAME]: {
+        image: "ipfs/go-ipfs",
+        pull_policy: "always",
+        ports: [`${IPFS_PORT}:${IPFS_PORT}`, "4001:4001"],
+        environment: {
+          IPFS_PROFILE: "server",
+        },
+        volumes: [`./${IPFS_CONTAINER_NAME}/:/container-init.d/`],
+      },
+      ...Object.fromEntries([
+        genNox({
+          name: bootStrapNoxName,
+          tcpPort: TCP_PORT_START,
+          webSocketPort: WEB_SOCKET_PORT_START,
+          bootStrapNoxName,
+        }),
+      ]),
+      ...Object.fromEntries(
+        restSecretKeys.map(({ name }, index) => {
+          return genNox({
+            name,
+            tcpPort: TCP_PORT_START + 1 + index,
+            webSocketPort: WEB_SOCKET_PORT_START + 1 + index,
+            bootStrapNoxName,
+          });
+        }),
+      ),
+    },
+    secrets: Object.fromEntries(
+      secretKeys.map(({ name, path: file }) => {
+        return [name, { file }];
+      }),
+    ),
+  };
+}
+
+async function genDefaultDockerCompose(
+  providerConfig: ProviderConfigReadonly,
+): Promise<GetDefaultConfig> {
+  const def = await genDockerCompose(providerConfig);
+
+  return () => {
+    return yamlDiffPatch("", {}, def);
+  };
+}
+
+const migrations: Migrations<Config> = [];
```

Comment:
maybe move all constant and type declarations to the top of the file?

it's pretty hard to follow when functions are interleaved with consts and type defs